diff --git a/src/main/java/io/github/johnjcool/keycloak/broker/cas/CasIdentityProvider.java b/src/main/java/io/github/johnjcool/keycloak/broker/cas/CasIdentityProvider.java
index 4e01883..11b9861 100644
--- a/src/main/java/io/github/johnjcool/keycloak/broker/cas/CasIdentityProvider.java
+++ b/src/main/java/io/github/johnjcool/keycloak/broker/cas/CasIdentityProvider.java
@@ -11,6 +11,7 @@ import jakarta.ws.rs.CookieParam;
 import jakarta.ws.rs.GET;
 import jakarta.ws.rs.Path;
 import jakarta.ws.rs.QueryParam;
+import jakarta.ws.rs.WebApplicationException;
 import jakarta.ws.rs.core.*;
 import jakarta.xml.bind.JAXBContext;
 import jakarta.xml.bind.JAXBException;
@@ -138,9 +139,13 @@ public class CasIdentityProvider extends AbstractIdentityProvider<CasIdentityPro
     public Response authResponse(
         @QueryParam(PROVIDER_PARAMETER_TICKET) final String ticket,
         @CookieParam(STATE_COOKIE_NAME) final Cookie stateCookie) {
-      return callback.authenticated(
-          getFederatedIdentity(
-              config, ticket, session.getContext().getUri(), stateCookie.getValue()));
+      try {
+        return callback.authenticated(
+            getFederatedIdentity(
+                config, ticket, session.getContext().getUri(), stateCookie.getValue()));
+      } catch (WebApplicationException e) {
+        return e.getResponse();
+      }
     }
 
     @GET
